shared/src/types/key/secp256k1.rs

```diff
@@ -316,7 +323,7 @@ impl<'de> Deserialize<'de> for Signature {
         struct ByteArrayVisitor;
 
         impl<'de> Visitor<'de> for ByteArrayVisitor {
-            type Value = [u8; libsecp256k1::util::SIGNATURE_SIZE];
+            type Value = [u8; libsecp256k1::util::SIGNATURE_SIZE + 1];
```

Comment:
this has probably already gone into main, but we could add a const here with a docstr explaining why this `+ 1` is needed